iothub/device/samples/convention-based-samples/TemperatureController/Parameter.cs

```diff
@@ -0,0 +1,86 @@
+        public string DeviceSecurityType { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_SECURITY_TYPE");
+
+        [Option(
+            'p',
+            "PrimaryConnectionString",
+            HelpText = "(Required if DeviceSecurityType is \"connectionString\"). \nThe primary connection string for the device to simulate." +
+            "\nDefaults to environment variable \"IOTHUB_DEVICE_CONNECTION_STRING\".")]
+        public string PrimaryConnectionString { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_CONNECTION_STRING");
+
+        [Option(
+            'e',
+            "DpsEndpoint",
+            HelpText = "(Required if DeviceSecurityType is \"dps\"). \nThe DPS endpoint to use during device provisioning." +
+            "\nDefaults to environment variable \"IOTHUB_DEVICE_DPS_ENDPOINT\".")]
+        public string DpsEndpoint { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_DPS_ENDPOINT");
+
+        [Option(
+            'i',
+            "DpsIdScope",
+            HelpText = "(Required if DeviceSecurityType is \"dps\"). \nThe DPS ID Scope to use during device provisioning." +
+            "\nDefaults to environment variable \"IOTHUB_DEVICE_DPS_ID_SCOPE\".")]
+        public string DpsIdScope { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_DPS_ID_SCOPE");
+
+        [Option(
+            'd',
+            "DeviceId",
+            HelpText = "(Required if DeviceSecurityType is \"dps\"). \nThe device registration Id to use during device provisioning." +
+            "\nDefaults to environment variable \"IOTHUB_DEVICE_DPS_DEVICE_ID\".")]
+        public string DeviceId { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_DPS_DEVICE_ID");
+
+        [Option(
+            'k',
+            "DeviceSymmetricKey",
+            HelpText = "(Required if DeviceSecurityType is \"dps\"). \nThe device symmetric key to use during device provisioning." +
+            "\nDefaults to environment variable \"IOTHUB_DEVICE_DPS_DEVICE_KEY\".")]
+        public string DeviceSymmetricKey { get; set; } = Environment.GetEnvironmentVariable("IOTHUB_DEVICE_DPS_DEVICE_KEY");
+
+        [Option(
+            'r',
+            "Application running time (in seconds)",
+            Required = false,
+            HelpText = "The running time for this console application. Leave it unassigned to run the application until it is explicitly canceled using Control+C.")]
+        public double? ApplicationRunningTime { get; set; }
+
+        public bool Validate(ILogger logger)
+        {
+            if (string.IsNullOrWhiteSpace(DeviceSecurityType))
+            {
+                logger.LogWarning("Device provisioning type not set, please set the environment variable \"IOTHUB_DEVICE_SECURITY_TYPE\"" +
+                    "or pass in \"-s | --DeviceSecurityType\" through command line. \nWill default to using \"dps\" flow.");
```

Comment:
Do we still default to dps?

Trying to recall what is the reasoning or was it the change only for bug bash?

Reply:
I'm not changing this part of the logic, since this is what the readme refers to. I am only focusing on updating the API calls.
If we want to update this flow, we can add a task and take it up later.